src/build/bundle.ts

```diff
@@ -176,6 +190,45 @@ export class Bundler {
     });
   }
 
+  _addSharedImportsToShell(bundles: Map<string, string[]>): string {
+    console.assert(this.shell != null);
+    let shellDeps = bundles.get(this.shell)
+        .slice()
+        .map((d) => path.relative(path.dirname(this.shell), d));
+    let shellDepsSet = new Set(shellDeps);
+    let file = this.streamResolver._files.get(this.shell);
+    console.assert(file != null);
+    let contents = file.contents.toString();
+    let doc = dom5.parse(contents);
+    let imports = dom5.queryAll(doc, dom5.predicates.AND(
+      dom5.predicates.hasTagName('link'),
+      dom5.predicates.hasAttrValue('rel', 'import')
+    ));
+    for (let _import of imports) {
```

Comment:
Comment on why you rip the imports out of the doc if they are in the shared deps bundle